impl/src/main/java/com/okta/sdk/impl/resource/OktaResourceHrefResolver.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.impl.resource;
+
+import com.okta.sdk.lang.Collections;
+import com.okta.sdk.resource.Resource;
+import com.okta.sdk.resource.application.AppUser;
+import com.okta.sdk.resource.application.Application;
+import com.okta.sdk.resource.group.Group;
+import com.okta.sdk.resource.group.rule.GroupRule;
+
+import java.util.Map;
+
+/**
+ * The default HREF resolver return a HAL self link if available ({@code _links.self.href}). If the {@code self} link is
+ * missing, it will be statically built for known objects (or null will be returned).
+ * <p>
+ * Known Objects:
+ * <ul>
+ *     <li>AppUser</li>
+ *     <li>Application</li>
+ *     <li>Group</li>
+ *     <li>GroupRule</li>
+ * </ul>
+ *
+ * @since 0.11.0
+ */
+public class OktaResourceHrefResolver implements ResourceHrefResolver {
+
+    private final ResourceHrefResolver halResourceHrefResolver = new HalResourceHrefResolver();
+
+    @Override
+    public <R extends Resource> String resolveHref(Map<String, ?> properties, Class<R> clazz) {
```

Comment:
Just a thought - Since we only return "self" href link, should this method be named `resolveSelfHref`? Will we ever need  "non-self" href links in the future?

Reply:
I think the `self` link is an implementation detail, in Okta's case we use HAL style links.  At Stormpath we used a root level property of `href`.